### PR TITLE
Add update object metadata API

### LINKED ISSUE #8251: Add API to update user metadata
## Objective

Provide an API to set user-defined metadata fields for an existing object.

## Out of Scope

1. Experimental API
1. Concurrency control - no need for conditional set operations
1. Support beyond a REST API and a means of using it through Python (no lakectl, UI or
   other SDK support for now)
1. Object user metadata diff presentation

### Swagger API

Create a new operation that allows replacing all metadata key-value pairs
for a given object.

1. Allows updating a single object's full set of user-defined metadata
1. Should mostly mimic the behavior of S3's
[PutObjectTagging](https://docs.aws.amazon.com/AmazonS3/latest/API/API_PutObjectTagging.html)
   API: Allowing to send a full set of tags for the given object to replace
   any existing tags, if any.

## RBAC Permissions

| Action       | Permission             | ARN    |
|--------------|------------------------|--------|
| Set object Metadata     | `fs:WriteObject`           | `arn:lakefs:fs:::repository/{repositoryId}/object/{objectKey}` |

### PR DESCRIPTION
**Pull request by arielshaqed**
This is an *experimental* feature.  When reviewing please comment about its usability, but focus mainly on correctness.

The code in the controller re-uses an existing physical address, which is supposedly a no-no.  However **this case is fine**: the physical address is used _by the same object being updated_.  So the only way for UGC to remove it would be to lose these races:
- Thread 1 gets object metadata
- Thread 2 deletes the object
- 👉🏽  **Thread 3 performs UGC and cleans up the physical object**
- Thread 1 writes a new entry for the object with the same physical address
Thread 3 is UGC so it only cleans up objects that were abandoned several hours ago.  Our UGC model allows us to lose the object in this (extreme) case - this is not a new lossage.

Fixes #8251.

### CODE REVIEW

**Comment by arielshaqed** [comment]
@ozkatz for API

(@N-o-Z for API and correctness and code and ..., of course)

File: api/swagger.yml [hunk 1]
```diff
@@ -4993,6 +5002,51 @@ paths:
...
+      operationId: updateObjectUserMetadata
+      summary: update object metadata
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/UpdateObjectUserMetadata"
+      responses:
+        201:
```
Comment (ozkatz) [inline api/swagger.yml:5037]: Is 201 Created customary for a PUT that updates an existing resource? (not asking that you'd change it, legitimately asking!)

File: cmd/lakectl/cmd/fs_update_metadata.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	"net/http"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/treeverse/lakefs/pkg/api/apigen"
+)
+
+var errExpectedKV = errors.New("expected <key>=<value>")
+
+var fsUpdateUserMetadataCmd = &cobra.Command{
```
Comment (ozkatz) [inline cmd/lakectl/cmd/fs_update_metadata.go:15]: Going above and beyond!

**Review by ozkatz — APPROVED** [review]
Looks great to me! Just wondering on 201 successful response for a PUT request

File: api/swagger.yml (hunk 1 above)
Comment (arielshaqed) [reply api/swagger.yml:5037]: [MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/PUT#successfully_creating_a_resource) says we're good:
> If the target resource does have a current representation and that representation is successfully modified with the state in the request, the origin server must send either a [200 OK](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/200) or a [204 No Content](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/204) to indicate successful completion of the request

Here we're always on the modification flow, you cannot PUT this is the object doesn't exist.

File: cmd/lakectl/cmd/fs_update_metadata.go (hunk 2 above)
Comment (arielshaqed) [reply cmd/lakectl/cmd/fs_update_metadata.go:15]: Easiest way to develop the code:-)

**Review by arielshaqed — COMMENTED** [review]
Thanks! 
@N-o-Z ... can you review or should I get someone else on this?

**Comment by arielshaqed** [comment]
@guy-har please review as I missed @N-o-Z' working hours 😭

File: api/swagger.yml [hunk 3]
```diff
@@ -4993,6 +5002,51 @@ paths:
...
+        description: path to object relative to the branch
+        required: true
+        schema:
+          type: string
+    put:
+      tags:
+        - objects
+        - experimental
+      operationId: updateObjectUserMetadata
+      summary: update object metadata
```
Comment (guy-har) [inline api/swagger.yml:5029]: Maybe we should mention this will override the current metadata

**Review by guy-har — APPROVED** [review]
LGTM 
Thanks!

File: api/swagger.yml (hunk 3 above)
Comment (arielshaqed) [reply api/swagger.yml]: Done, thanks!

**Review by arielshaqed — COMMENTED** [review]
Thanks, pulling...

**Pull request merged**